Standardize resource joining docs around the Proxy (#22111)
gravitational/teleport: docs/pages/management/register-services/register-services-auth-service.mdx

presents the token, indicating to your Teleport cluster that it is trustworthy.

<Admonition type="warning" title="Before following this guide">

In most cases, you should be able to register Teleport services via the Proxy
Service using a join token or another method. See our overview of [Registering
Services](../registering-services.mdx) for all available methods.

Only connect services directly to the Auth Service if no other methods are
suitable, as we recommend exposing the Auth Service to as few sources of ingress
traffic as possible.

</Admonition>

## Prerequisites

(!docs/pages/includes/edition-prereqs-tabs.mdx!)

- A Linux server that you will use to host your Teleport service. 

  In this guide, we will show you how to register a Teleport SSH Service
  instance. This approach applies to new Proxy Service instances as well as the
  Kubernetes Service, Database Service, and other Teleport services for
  accessing resources in your infrastructure.

(!docs/pages/includes/tctl.mdx!)

## Step 1/3. Install Teleport

Install Teleport on the host where you will run your Teleport service.

(!docs/pages/includes/install-linux.mdx!)

> ## Step 2/3. Join your service to the cluster

Reviewer: Technically, `instance` would be a more correct term, as multiple services can run in a single Teleport process.

In this section, we will join your Teleport service to the Teleport cluster by:

> - Obtaining a CA pin to establish trust with the Auth Service

Reviewer: CA pinning is only used when connecting directly to auth. If we're recommending joining via proxy then I'd probably remove this step.

Author: I've split the original "Joining Nodes" guide to have separate guides for joining via the Auth/Proxy. This way, we don't need to have as much branching within the guide. This guide has to do with the Auth Service.

### Generate a token

> Teleport only allows access to services that have joined the cluster.

Reviewer: ```suggestion
Teleport only allows access to resources that have joined the cluster.
```

Once a service joins, it receives a host certificate signed by the cluster's
Auth Service. To receive a host certificate upon joining a cluster, a new
Teleport host must present an **join token**.

> An join token also defines which role a new host can assume within a cluster:

Reviewer: ```suggestion
A join token also defines which role a new host can assume within a cluster:
```

following example, a new token is created with a TTL of five minutes:

```code

> $ export INVITE_TOKEN=$(tctl nodes add --ttl=5m --roles=node | grep "join token:" | grep -Eo "[0-9a-z]{32}")

Reviewer: Maybe we should use `tctl tokens add` instead. This was originally written when SSH nodes were the only type of supported resource.

$ echo $INVITE_TOKEN
# (=presets.tokens.first=)

# You can also list all generated non-expired tokens:
$ tctl tokens ls
# Token                            Type            Expiry Time
# ------------------------         -----------     ---------------

> # (=presets.tokens.first=)         service            25 Sep 18 00:21 UTC

Reviewer: `service` would never appear under the Type column like this.

```code
$ tctl nodes add --ttl=5m --roles=node,proxy --token=secret-value

> # The join token: secret-value

Reviewer: Unless you've actually changed the tctl source code I would remove this.

   --auth-server=${ADDR?}
```

> As new services come online, they start sending ping requests every few seconds to

Reviewer: ```suggestion
As new services come online, they start sending heartbeats every few seconds to
```

the Auth Service. This allows users to explore cluster membership and size.

> Run the following command on your local machine to see all of the Teleport services

Reviewer: `tctl nodes ls` only shows SSH nodes, so the docs are now incorrect here.

in your cluster:

```code
$ tctl nodes ls

> service Name     service ID                                  Address            Labels

Reviewer: Revert this.

# Token                                Role       Expiry Time (UTC)
# -----                                ----       -----------------
# (=presets.tokens.first=)     Proxy      never

> # (=presets.tokens.second=)    Service      17 May 16 03:51 UTC

Reviewer: Revert

# (=presets.tokens.third=)     Signup     17 May 16 04:24 UTC
```

<Admonition type="tip" title="Signup tokens">

The output of `tctl tokens ls` includes tokens used for adding users alongside
tokens used for adding services to your cluster.

</Admonition>

In this example, the first token has a `never` expiry date because it is a
static token configured via a config file.

> The token with the `service` role was generated to invite a new service to this

Reviewer: There is no `service` role. Revert this.
